### PR TITLE
Uniform testing update

### PR DESCRIPTION
**Pull request by timbussmann**
Changing the `TestableUniformSession` implementation to allow inner test contexts like `TestableMessageHandlerContext` or `TestableMessageSession` to allow similar testing scenarios as with the deprecated fluent-style API where all calls to either `IUniformSession` as well as any other pipeline API (e.g. `IMessageHandlerContext`) or `IMessageSession` share a common state, as they would in practice once the uniform session is resolved.

See the added unit tests on some examples how this might be used when doing some more complex unit testing scenarios where both context APIs as well as `IUniformSession` APIs (typically further down the code path where shared code components are used) are called.

### CODE REVIEW

File: src/NServiceBus.UniformSession.Testing.Tests/.editorconfig [hunk 1]
```diff
@@ -1,4 +1,8 @@
 [*.cs]
 
+# Justification: Test project
+dotnet_diagnostic.CA2007.severity = none
```
Comment (timbussmann) [inline src/NServiceBus.UniformSession.Testing.Tests/.editorconfig:4]: same values as in the core editorconfig for unit tests

File: src/NServiceBus.UniformSession.Testing/TestableUniformSession.cs [hunk 2]
```diff
@@ -5,48 +5,39 @@
     using System.Threading.Tasks;
     using NServiceBus.Testing;
 
-    [System.Diagnostics.CodeAnalysis.SuppressMessage("Code", "PS0002:Instance methods on types implementing ICancellableContext should not have a CancellationToken parameter", Justification = "Testable object implementing interface explicitly")]
-    public class TestableUniformSession : TestablePipelineContext, IUniformSession
+    public class TestableUniformSession : IUniformSession
```
Comment (timbussmann) [inline src/NServiceBus.UniformSession.Testing/TestableUniformSession.cs:8]: this type no longer directly inheriting from `TestablePipelineContext` might be a major change, but from my understanding this seems to be irrelevant in practice because you'd basically never directly use it as a `TestablePipelineContext` and I guess this inheritance has only been setup reuse existing testing-implementations.

File: src/NServiceBus.UniformSession.Testing.Tests/TestableUniformSessionTests.cs [hunk 3]
```diff
@@ -5,17 +5,67 @@
...
+            Assert.AreEqual(2, handlerContext.SentMessages.Length);
+
+            Assert.IsTrue(handlerContext.DoNotContinueDispatchingCurrentMessageToHandlersWasCalled);
+        }
+
+        [Test]
+        public async Task Shares_state_with_pipeline_context()
+        {
+            var pipelineContext = new TestableIncomingLogicalMessageContext();
+            var uniformSession = new TestableUniformSession(pipelineContext);
```
Comment (timbussmann) [inline src/NServiceBus.UniformSession.Testing.Tests/TestableUniformSessionTests.cs:34]: this approach allows passing any testable pipeline context from any stage rather than limiting it to some stages.

**Review by andreasohlund — APPROVED** [review]

**Review by danielmarbach — APPROVED** [review]

File: src/NServiceBus.UniformSession.Testing.Tests/TestableUniformSessionTests.cs [hunk 4]
```diff
@@ -45,43 +95,86 @@ public ReusableComponent(IUniformSession session)
...
 
-            public SomeHandler(IUniformSession session) => this.session = session;
+            public MessageSessionComponent(IUniformSession uniformSession)
+            {
+                this.uniformSession = uniformSession;
+            }
 
-            public Task Handle(SomeCommand message, IMessageHandlerContext context)
-                => session.Publish(new SomeEvent());
+#pragma warning disable PS0018 // A task-returning method should have a CancellationToken parameter unless it has a parameter implementing ICancellableContext
```
Comment (danielmarbach) [inline src/NServiceBus.UniformSession.Testing.Tests/TestableUniformSessionTests.cs:148]: Can we move this to editorconfig?

**Review by danielmarbach — APPROVED** [review]

File: src/NServiceBus.UniformSession.Testing.Tests/TestableUniformSessionTests.cs [hunk 5]
```diff
@@ -45,43 +95,86 @@ public ReusableComponent(IUniformSession session)
...
+                await messageSession.Send<ISomeOtherCommand>(_ => { });
+                await messageSession.Publish<ISomeOtherEvent>(_ => { });
+                await uniformSession.Send<ISomeOtherCommand>(_ => { });
+                await uniformSession.Publish<ISomeOtherEvent>(_ => { });
+            }
         }
 
-        class SomeSagaData : ContainSagaData
+        // Disable warning because public type is necessary for interface messages
+        // ReSharper disable once MemberCanBePrivate.Global
```
[the remark this answers is not in the document]
Comment (timbussmann) [reply src/NServiceBus.UniformSession.Testing.Tests/TestableUniformSessionTests.cs]: removed

File: src/NServiceBus.UniformSession.Testing.Tests/TestableUniformSessionTests.cs (hunk 4 above)
Comment (timbussmann) [reply src/NServiceBus.UniformSession.Testing.Tests/TestableUniformSessionTests.cs]: done
